src/pages/storage/forms/DuplicateVolumeForm.tsx

```diff
@@ -0,0 +1,279 @@
+        controllerState,
+        `storage-pools/${pool}/volumes/custom`,
+      );
+
+      if (notFound) {
+        return true;
+      }
+
+      return this.createError({
+        path: "name",
+        message:
+          "A volume with this name already exist in the target project and storage pool",
+      });
+    });
+
+  const formik = useFormik<LxdStorageVolumeDuplicate>({
+    initialValues: {
+      name: getDuplicatedVolumeName(volume),
+      project: volume.project,
+      volumeOnly: false,
+      member: isClustered ? volume.location : "",
+      pool: volume.pool,
+    },
+    enableReinitialize: true,
+    validationSchema,
+    onSubmit: (values) => {
+      const payload: Partial<LxdStorageVolume> = {
+        name: values.name,
+        type: "custom",
+        config: volume.config,
+        description: volume.description,
+        content_type: volume.content_type,
+        source: {
+          name: volume.name,
+          type: "copy",
+          pool: volume.pool,
+          volume_only: values.volumeOnly,
+          // logic from the lxc source code.
+          // We should not set source.project if target project is the same as the source project
+          project:
+            values.project !== volume.project ? volume.project : undefined,
+          location: volume.location,
+        },
+      };
+
+      const existingVolumeLink = (
+        <Link
+          to={`/ui/project/${volume.project}/storage/pool/${volume.pool}/volumes/custom/${volume.name}`}
+          onClick={(e) => e.stopPropagation()}
+        >
+          <ItemName item={volume} />
+        </Link>
+      );
+
+      const targetProject =
+        values.project === formik.initialValues.project ? "" : values.project;
+
+      const targetMember =
+        values.member === formik.initialValues.member ? "" : values.member;
+
+      duplicateStorageVolume(payload, values.pool, targetProject, targetMember)
+        .then((operation) => {
+          toastNotify.info(`Duplication of volume ${volume.name} started.`);
+          eventQueue.set(
+            operation.metadata.id,
+            () => notifySuccess(values.name, values.project, values.pool),
+            (msg) =>
+              toastNotify.failure(
+                "Volume duplication failed.",
+                new Error(msg),
+                existingVolumeLink,
+              ),
+          );
+        })
+        .catch((e) => {
+          toastNotify.failure(
+            "Volume duplication failed.",
+            e,
+            existingVolumeLink,
+          );
+        })
+        .finally(() => {
+          close();
+        });
+    },
+  });
+
+  const disableClusterMemberSelector =
+    formik.values.project !== volume.project
+      ? "Can not migrate cluster member when changing project"
+      : undefined;
+
+  const disableProjectSelector =
+    formik.values.member !== volume.location
+      ? "Can not change project when migrating cluster member"
+      : undefined;
```

Comment:
This doesn't work in the unclustered case, because it disables it there as well.

Reply:
good catch, the logic should be `isClustered && formik.values.member !== volume.location`